wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecord.g4

```diff
@@ -77,10 +78,14 @@ opt:
     | defaultval
     | xpath
     | xmldefaultval
+    | blob
 ;
 maxlength:
-    'MAXLENGTH' OPAREN INT CPAREN
+    ('MAXLENGTH' OPAREN INT CPAREN | 'maxlength' OPAREN INT CPAREN | 'maxLength' OPAREN INT CPAREN)
 ;
+
+blob: ('BLOB' | 'blob');
```

Comment:
consider implementing a case insensitive grammar:
https://github.com/antlr/antlr4/blob/master/doc/case-insensitive-lexing.md

Reply:
I tried using a case insensitive lexer override class when I first stood up the ecl lexer. The case insensitive stream wasn't the best solution as we want to retain some lower case things: filenames in a dataset attribute, TERMINATOR('\n'), etc.

The other option, defining fragments for each letter, I hadn't tried. I just tried it for BLOB and several of the parsing cases now fail. I could probably figure out what's up but it would take a while .

Reply:
Fair enough, I'd put it off